src/pages/Explorer/Aside.svelte

```diff
@@ -5,20 +5,48 @@
   import SocialTrend from './Layouts/SocialTrend.svelte'
   import WeeklyReport from './Layouts/WeeklyReport.svelte'
   import SheetsTemplate from './Layouts/SheetsTemplate.svelte'
+  import { queryExplorerItems, getReports, getTemplates } from './api'
+  import { EntityType } from './const'
 
   let className = ''
   export { className as class }
 
-  // TODO: add getItems props for each widget
+  const PAGE_SIZE = 5
+  
+  const getRecentItems = (type, key) => (page) =>
+    queryExplorerItems({ types: [type], page, pageSize: PAGE_SIZE }).then(({ pages, items }) => ({
+      pages,
+      items: items.map((item) => item[key]),
+    }))
+
+  const getCustomItems = getFunc => page => getFunc().then(res => {
```

Comment:
1. Better to rename the `getFunc` to `queryItems`;
2. `res` is a vague name, better to use `items` or something like this